x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamTransportAction.java

```diff
@@ -56,6 +56,10 @@ public ReindexDataStreamTransportAction(
 
     @Override
     protected void doExecute(Task task, ReindexDataStreamRequest request, ActionListener<ReindexDataStreamResponse> listener) {
+        if (request.getMode().equals("upgrade") == false) {
+            listener.onFailure(new IllegalArgumentException("Only mode 'upgrade' is supported"));
+            return;
+        }
```

Comment:
If we make it an enum, we can move this into the request parsing, so that it doesn't go over the transport before being rejected